File: src/frontend/contabil/lib/app/data/model/plano_centro_resultado_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:contabil/app/infra/infra_imports.dart';
import 'package:intl/intl.dart';

class PlanoCentroResultadoModel {
	int? id;
	String? nome;
	String? mascara;
	int? niveis;
	DateTime? dataInclusao;

	PlanoCentroResultadoModel({
		this.id,
		this.nome,
		this.mascara,
		this.niveis,
		this.dataInclusao,
	});

	static List<String> dbColumns = <String>[
		'id',
		'nome',
		'mascara',
		'niveis',
		'data_inclusao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Nome',
		'Mascara',
		'Niveis',
		'Data Inclusao',
	];

	PlanoCentroResultadoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		nome = jsonData['nome'];
		mascara = jsonData['mascara'];
		niveis = jsonData['niveis'];
		dataInclusao = jsonData['dataInclusao'] != null ? DateTime.tryParse(jsonData['dataInclusao']) : null;
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['nome'] = nome;
		jsonData['mascara'] = mascara;
		jsonData['niveis'] = niveis;
		jsonData['dataInclusao'] = dataInclusao != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataInclusao!) : null;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		nome = plutoRow.cells['nome']?.value;
		mascara = plutoRow.cells['mascara']?.value;
		niveis = plutoRow.cells['niveis']?.value;
		dataInclusao = Util.stringToDate(plutoRow.cells['dataInclusao']?.value);
	}	

	PlanoCentroResultadoModel clone() {
		return PlanoCentroResultadoModel(
			id: id,
			nome: nome,
			mascara: mascara,
			niveis: niveis,
			dataInclusao: dataInclusao,
		);			
	}

	
}

File: backend/contabil/src/model/plano_centro_resultado_model.py
from src import db


class PlanoCentroResultadoModel(db.Model):
    __tablename__ = 'plano_centro_resultado'

    id = db.Column(db.Integer, primary_key=True)
    nome = db.Column(db.String(100))
    mascara = db.Column(db.String(50))
    niveis = db.Column(db.Integer)
    data_inclusao = db.Column(db.DateTime)


    def mapping(self, data):
        self.id = data.get('id')
        self.nome = data.get('nome')
        self.mascara = data.get('mascara')
        self.niveis = data.get('niveis')
        self.data_inclusao = data.get('dataInclusao')

    def serialize(self):
        return {
            'id': self.id,
            'nome': self.nome,
            'mascara': self.mascara,
            'niveis': self.niveis,
            'dataInclusao': self.data_inclusao.isoformat(),
        }